src/main/java/org/openrewrite/maven/AbstractRewriteBaseRunMojo.java

```diff
@@ -126,6 +148,62 @@ protected ResultsContainer listResults(ExecutionContext ctx) throws MojoExecutio
         }
     }
 
+    private void configureRecipeOptions(Recipe recipe) throws MojoExecutionException {
+        if (recipe instanceof CompositeRecipe) {
+            CompositeRecipe composite = (CompositeRecipe) recipe;
+            for (Recipe child : composite.getRecipeList()) {
+                this.configureRecipeOptions(child);
+            }
+        } else if (recipe instanceof DeclarativeRecipe) {
+            DeclarativeRecipe declarativeRecipe = (DeclarativeRecipe) recipe;
+            for (Recipe child : declarativeRecipe.getRecipeList()) {
+                this.configureRecipeOptions(child);
+            }
+        } else {
```

Comment:
From the discussion on the issue I think it's best that we don't try to support Declarative or Composite recipes just yet.
- https://github.com/openrewrite/rewrite-maven-plugin/issues/799

These would be hierarchical, possibly contain the same recipe multiple times, as well as other recipes that do not have the same options. As such they are highly likely to trigger exceptions when trying to `updateOption` below.

In terms of how to fail I suggest we validate early that the recipe is not a Composite/Declarative recipe when any options are set.

Reply:
Makes sense. I can undo this